vane-network (#2052)
w3f/Grants-Program: applications/vane-network.md

 - A safety non delivered xcm transaction handler

![](https://github.com/2-5-Foundation/vane/blob/main/artifacts/VaneComponents%20for%20grant.jpeg?raw=true)

*NOTE* vane-network is not building an sdk, it will be a fully fledged ready to use product

### Ecosystem Fit

Help us locate your project in the Polkadot/Substrate/Kusama landscape and what problems it tries to solve by answering each of these questions:

**Where and how does your project fit into the ecosystem?**

  - This project is a clear demonstration on how XCM can be used to introduce safety and improve UX in web3 ecosystem.

  - And XCM can be used to add a sort of extension layer ontop of existing network and introduce a product layer.

**Who is your target audience (parachain/dapp/wallet/UI developers, designers, your own user base, some dapp's userbase, yourself)?**
   - Crypto users for transactions
   - Social commerce entrepreneurs

**What need(s) does your project meet?**
  - Introducing safety for web3 mass adoption
  - Bridging and using web3 based technology to grow online businesses.

**Are there any other projects similar to yours in the Substrate / Polkadot / Kusama ecosystem?**
  

>   - No, but ENS and Naing services try to solve the issue of ensuring wrong address mistakes are resolved. But Vane ensures this product to work just outside the box without purchasing any name , just existig addressbut introducing confirmation and reverting system ontop of existing networks.

Reviewer: ```suggestion
  - No, but ENS and Naing services try to solve the issue of ensuring wrong address mistakes are resolved. But Vane ensures this product to work just outside the box without purchasing any name, just existing address but introducing confirmation and reverting system on top of existing networks.
```

### Milestone 2 - Pallet contract & chain extensions for extending vane-product services.

### Notion templates marketplace as a POC

- **Estimated duration:** 1.2 month
- **FTE:**  2.65
- **Costs:** 11,000 USD

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| **0a.** | License | Apache 2.0
| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |

> | 0e. | Video tutorial | We will publish an workshop on X/Twitter that explains [...] (what was done/achieved as part of the grant and how the product should be used) |

Reviewer: Appreciate the video tutorial, however, could you also add an article, at least to milestone 2? See also the [mandatory deliverables](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality) in the application template.

Author: We thought it will be better to not include an article and to be more effective on the video for better engagement and get proper feedback from users/community.

Reviewer: Ok I'm fine with that. But could you add a remark to the doc itself, where you explain this?

Reviewer: This has not been addressed yet. Also, there is still some leftover text present that was copied over from the template:

> [...] (what was done/achieved as part of the grant and how the product should be used)

| 2. | permissioned pallet contract & chain extensions | A permissioned wasm contract pallet and chain exensions for vane risk free transfer system to be used to extend Vane product services. Account who intend to extend vane product services must apply to be whitelisted to deploy the contract.<br><br>**Chain extensions**<br> This will be limited to vane specific pallets extrinsics only, which includes, pallet_vane_xcm_transfer, pallet_vane_relay_asset, pallet_transfer_claim |
| 3. | Notion template marketplace contract| The marketplace will showcase how we can enable transaction customization for social-commerce business and foster growth. It will contain, listing of notion templates, securely limiting access of the notion template, buying of the template, Order placement and tracking.|
| 4. | Backend (Nest JS) | All chain ( vane, polkadot/rococo, assethub) api interaction implementation, API chart workflow, notion marketplace implementation for storing notion pages, notion users as a POC |

> | 5. | Frontend (Next JS) |  A front end prototype in Figma, A front end interaction for implemented feature hosted in vercel which will be only for wide screen ( no responsiveness ) as later on (not this grant ) we will implement a mobile application |

Reviewer: Usually we require wireframes to be delivered as part of the grant proposal. You can use any suitable tool for that (incl. figma).

Author: The wireframe as product clickable prototype will be submitted

Reviewer: Just to be clear - we expect the wireframes to be delivered before we approve the proposal. It doesn't necessarily have to be clickable.

Reviewer: wireframes have been added, resolving this comment

# vane-network

- **Team Name:** Vane-Network
- **Payment Address:** 14uNH42NjxS3c4VSxVyHjZYea5ZyJEYWYxjRzmxKQB75pdRB (AssetHub - USDT)
- **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):**  2
- **Email:** MrishoLukamba@proton.me || abdulrazzaqlukamba@gmail.com ||

## Project Overview :page_facing_up:

### Overview

- **Beneficiary** Mrisho Lukamba
  

Vane is a parachain that act as an extension product and safety layer for Polkadot and AssetHub which provides services to help trust and growth on social commerce business and introducing transaction safety in web3.

Vane-network consist of the following functionalities

 - Confirmation and reversible transactions
 - Customizable and trustless transactions for social-commerce business

>  - Native token-less parachain enabling usage of DOT, USDT & USDT for all interactions via XCM

Reviewer: Does this mean your solution enables non-native (e.g. USDT) transfers without the necessity to own the native asset? I believe this isn't included in the milestones yet, could you fix that?

### Milestone 1 — Trasanction Delivery Tracking System with XCM & Pallet Tansaction Claim.
 

- **Estimated duration:** 0.7 month
- **FTE:**  2.5
- **Costs:** 9,000 USD

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| **0a.** | License | Apache 2.0
| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
| 0e. | Video tutorial | We will publish an workshop on X/Twitter that explains [...] (what was done/achieved as part of the grant and how the product should be used) |

> | 1. | Transaction Delivery Tracking System | As vane-risk free transfer system involves the following parts.<br> Relay_chain xcm reserveTransfer to Vane, confirmation of sender and receiver's address,sending xcm instructions to relay/assethub for transacting asset from vane para account to receiver.<br> The following are notable issues which may arrive; Failure to deliver and execute XCM message to vane, and relay/assethub.<br><br> A tracking functionality of where the transaction stuck is essential and will be implemented as a struct of an enum and stored in transaction tracking storage with key of multi_id.<br> The updating will be based on queries of effect of what the transaction is supposed to do using XCM.<br>|

Reviewer: Is this a pallet? Pls specify directly in the doc.

Author: Yes it is a pallet

### Milestone 2 - Pallet contract & chain extensions for extending vane-product services.

### Notion templates marketplace as a POC

- **Estimated duration:** 1.2 month
- **FTE:**  2.65
- **Costs:** 11,000 USD

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| **0a.** | License | Apache 2.0
| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
| 0e. | Video tutorial | We will publish an workshop on X/Twitter that explains [...] (what was done/achieved as part of the grant and how the product should be used) |

> | 1. | pallet transaction claim resolver | We will create a Substrate module that will handle failed transaction based on XCM delivery issues. Allowing users to claim their incomplete transactions|

Reviewer: Could you outline the supported reasons for xcm delivery failure here?

> | 2. | permissioned pallet contract & chain extensions | A permissioned wasm contract pallet and chain exensions for vane risk free transfer system to be used to extend Vane product services. Account who intend to extend vane product services must apply to be whitelisted to deploy the contract.<br><br>**Chain extensions**<br> This will be limited to vane specific pallets extrinsics only, which includes, pallet_vane_xcm_transfer, pallet_vane_relay_asset, pallet_transfer_claim |

Reviewer: > Account who intend to extend vane product services must apply to be whitelisted to deploy the contract.

Do you refer to smart contracts here or to some kind of financial contract? If smart contract, does that mean each user has to deploy a smart contract in order to use vane?

Author: No user can extend vane functionalities by deploying a smart contract for an application that uses vane chain extensions and base layer functionality ( risk free and trustless transfers)

> | 3. | Notion template marketplace contract| The marketplace will showcase how we can enable transaction customization for social-commerce business and foster growth. It will contain, listing of notion templates, securely limiting access of the notion template, buying of the template, Order placement and tracking.|

Reviewer: Did I understand correctly - this is a marketplace to trade notion.so templates?
Could you please list the tech used here?

Author: Yes it is, and the tech stack used I listed above. TS, Nest Js, Next Js, Rust Ink Smart contract. ( showcasing extending vane functionality utilizing risk free transfers

| 4. | Backend (Nest JS) | All chain ( vane, polkadot/rococo, assethub) api interaction implementation, API chart workflow, notion marketplace implementation for storing notion pages, notion users as a POC |

> | 5. | Frontend (Next JS) |  A front end prototype in Figma, A front end interaction for implemented feature hosted in vercel which will be only for wide screen ( no responsiveness ) as later on (not this grant ) we will implement a mobile application |

Reviewer: ```suggestion
| 5. | Frontend (Next JS) |  Front end wireframes in Figma and a NextJs-based front end for the implemented feature hosted in vercel which will be only for wide screen (no responsiveness) as later on (not this grant) we will implement a mobile application |
```

Reviewer: This would make it clearer that it's an actual implementation, not just figma wireframes.

Author: We did mention that it will figma wireframes and the actual front end
